samples/server/go-server/api/server/storage/embeddedstore.go

```diff
@@ -0,0 +1,415 @@
+// Copyright 2017 The Grafeas Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/boltdb/bolt"
+	"github.com/golang/protobuf/proto"
+	"github.com/grafeas/grafeas/samples/server/go-server/api/server/name"
+	"github.com/grafeas/grafeas/server-go"
+	pb "github.com/grafeas/grafeas/v1alpha1/proto"
+	opspb "google.golang.org/genproto/googleapis/longrunning"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"log"
+	"os"
+	"path/filepath"
+)
+
+const (
+	bucketOccurrences = "occurrences"
+	bucketProjects    = "projects"
+	bucketNotes       = "notes"
+	bucketOperations  = "operations"
+)
+
+var (
+	errKeyExists = fmt.Errorf("key exists")
+	errNoKey     = fmt.Errorf("key missing")
+)
+
+type EmbeddedStoreConfig struct {
+	Path string `yaml:"path"` // Path is the folder path to storage files
+}
+
+// embeddedStore is a storage solution for Grafeas based on boltdb
+type embeddedStore struct {
+	db *bolt.DB
+}
+
+// NewEmbeddedStore creates a embeddedS store with initialized filesystem
+func NewEmbeddedStore(config *EmbeddedStoreConfig) server.Storager {
+	if err := os.MkdirAll(config.Path, 0700); err != nil {
+		log.Fatalf("Failed to create config directory %v", err)
+	}
+	db, err := bolt.Open(filepath.Join(config.Path, "grafeas.db"), 0600, nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+	if err := db.Update(func(tx *bolt.Tx) error {
+		if _, err := tx.CreateBucketIfNotExists([]byte(bucketOccurrences)); err != nil {
+			return err
+		}
+		if _, err := tx.CreateBucketIfNotExists([]byte(bucketProjects)); err != nil {
+			return err
+		}
+		if _, err := tx.CreateBucketIfNotExists([]byte(bucketNotes)); err != nil {
+			return err
+		}
+		if _, err := tx.CreateBucketIfNotExists([]byte(bucketOperations)); err != nil {
+			return err
+		}
+		return nil
+	}); err != nil {
+		log.Fatal(err)
+	}
+	return &embeddedStore{db: db}
+}
+
+// CreateProject adds the specified project to the embedded store
+func (m *embeddedStore) CreateProject(pID string) error {
+	err := m.update(bucketProjects, pID, true, &pb.Project{Name: name.FormatProject(pID)})
+	if err == errKeyExists {
+		return status.Error(codes.AlreadyExists, fmt.Sprintf("Project with name %q already exists", pID))
```

Comment:
You can use status.Errorf for all of these.

Reply:
Thanks @vtsao. I've replaced all usages of `status.Error` with `status.Errorf` for `embeededStore` `pgSQLStore` and `memStore`.